### PR TITLE
Add support for verify functionality.

### PR DESCRIPTION
**Pull request by sa-bpelakh**
* SPARQL and SHACL validation. 
* Transform via SPARQL Update query.
* Check return code of Java tool - this would allow us to run something like a reasoner, and abort if the model is not consistent.

I added some rudimentary unit tests, here's the coverage report (increase from 58% to 63%).

```
Name                     Stmts   Miss  Cover
--------------------------------------------
onto_tool\__init__.py        0      0   100%
onto_tool\mdutils.py        21     13    38%
onto_tool\onto_tool.py     549    166    70%
onto_tool\ontograph.py      68     55    19%
--------------------------------------------
TOTAL                      638    234    63%
```

I added to the documentation, for which @rjyounes will surely have (much appreciated, really) suggestions. ✍️ 
The unit tests serve as pretty good examples of the supported capability as well.

I replaced the JSON-formatted bundle JSON schema with a YAML version, because it's more streamlined and reads better. I will comment on the new sections, as most of it is unchanged.

### CODE REVIEW

File: onto_tool/bundle_schema.yaml [hunk 1]
```diff
@@ -0,0 +1,323 @@
...
+            type: object
+            properties:
+              source:
+                $ref: '#/definitions/non_empty_string'
+              includes:
+                $ref: '#/definitions/include_list'
+            required:
+              - source
+        required:
+          - queries
```
Comment (sa-bpelakh) [inline onto_tool/bundle_schema.yaml:65]: This was added to support specifying either `query` or a list via `queries` for SELECT and ASK Verification.

File: onto_tool/bundle_schema.yaml [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+                      properties:
+                        source:
+                          $ref: '#/definitions/non_empty_string'
+                        includes:
+                          $ref: '#/definitions/include_list'
+                  required:
+                    - shapes
+            required:
+              - type
+              - source
```
Comment (sa-bpelakh) [inline onto_tool/bundle_schema.yaml:188]: This is the schema for the new `verify` action. `type` and `source` are mandatory, the other options depend on `type`.

File: onto_tool/bundle_schema.yaml [hunk 3]
```diff
@@ -0,0 +1,336 @@
...
+          then:
+            allOf:
+              - $ref: '#/definitions/multi_query_spec'
+              - properties:
+                  format:
+                    type: string
+                    enum:
+                      - turtle
+                      - xml
+                      - nt
```
Comment (sa-bpelakh) [inline onto_tool/bundle_schema.yaml:123]: Modified this definition to allow for SPARQL Update transformation.

File: README.md [hunk 4]
```diff
@@ -257,3 +279,40 @@ one of the following values:
...
+        query: >
+          prefix skos: <http://www.w3.org/2004/02/skos/core#>
+          select ?unlabeled where {{
+            ?unlabeled a ?type .
+            filter not exists {{ ?unlabeled skos:prefLabel ?label }}
+          }}
+        ```
+      * If `queries` is provided, a list of queries will be built from the `source` and `includes`
+        sub-options.
+    * If `type` is `ask`, one or more SPARQL `ASK` queries will be executed, and if any of them return
```
Comment (rjyounes) [inline README.md:302]: It would be good to provide an example of the ask query so we know how expected should be formulated.

**Review by rjyounes — COMMENTED** [review]
One minor suggestions.

File: README.md (hunk 4 above)
Comment (sa-bpelakh) [reply README.md:303]: Added.

**Comment by sa-bpelakh** [comment]
Added some improvements based on comments from @uscholdm, most importantly a `stopOnFail` option that allows for all tests to run regardless of success before exiting. Coverage is a little better now with more testing.

```
----------- coverage: platform win32, python 3.7.4-final-0 -----------
Name                     Stmts   Miss  Cover
--------------------------------------------
onto_tool\__init__.py        0      0   100%
onto_tool\mdutils.py        21     13    38%
onto_tool\onto_tool.py     582    152    74%
onto_tool\ontograph.py      68     55    19%
--------------------------------------------
TOTAL                      671    220    67%
```

**Comment by rjyounes** [comment]
Good idea!

**Pull request merged**